<launch>
    <arg name="namespace" default="tracking" />
    <arg name="tracker_input_topic" default="/detection/objects" />
    <arg name="tracker_output_topic" default="/tracking/objects" />
    <arg name="gating_threshold" default="1.8" />
    <arg name="life_time_threshold" default="5" />
    <arg name="static_num_history_threshold" default="3" />
    <arg name="static_velocity_threshold" default="0.5" />
    <arg name="merge_distance_threshold" default="0.5" />

    <!--visualization parameter-->
    <arg name="objects_markers_topic" default="/objects_markers" />
    <arg name="objects_topic" default="/objects" />
    <arg name="object_speed_threshold" default="0.1" />
    <arg name="arrow_speed_threshold" default="0.25" />
    <arg name="marker_display_duration" default="0.2" />
    <arg name="label_color" default="{255.,255.,255.,1.0}" />
    <arg name="arrow_color" default="{0.,255.,0.,0.8}" />
    <arg name="hull_color" default="{51.,204.,51.,0.8}" />
    <arg name="box_color" default="{51.,128.,204.,0.8}" />
    <arg name="centroid_color" default="{77.,121.,255.,0.8}" />

    <!-- Include detection launch file -->
    <include file="$(find detection)/launch/euclideanClustering.launch" >
        <arg name="objects_markers_topic" value="$(arg objects_markers_topic)" />
        <arg name="objects_topic" value="$(arg objects_topic)" />
        <arg name="object_speed_threshold" value="$(arg object_speed_threshold)" />
        <arg name="arrow_speed_threshold" value="$(arg arrow_speed_threshold)" />
        <arg name="marker_display_duration" value="$(arg marker_display_duration)" />
        <arg name="label_color" value="$(arg label_color)" />
        <arg name="arrow_color" value="$(arg arrow_color)" />
        <arg name="hull_color" value="$(arg hull_color)" />
        <arg name="box_color" value="$(arg box_color)" />
        <arg name="centroid_color" value="$(arg centroid_color)" />
    </include>

    <group ns="$(arg namespace)">
        <remap from="objects_in" to="$(arg tracker_input_topic)" />
        <remap from="objects_out" to="$(arg tracker_output_topic)" />
        
        <!-- Tracker -->
        <node pkg="tracking" type="tracking" name="kfTracker" output="screen">
            <param name="gating_threshold" value="$(arg gating_threshold)" />
            <param name="life_time_threshold" value="$(arg life_time_threshold)" />
            <param name="static_num_history_threshold" value="$(arg static_num_history_threshold)" />
            <param name="static_velocity_threshold" value="$(arg static_velocity_threshold)" />
            <param name="merge_distance_threshold" value="$(arg merge_distance_threshold)" />
        </node> 

        <node pkg="visualization" type="visualize_perception_node" name="visualize_perception_node" output="screen">
            <param name="objects_markers_topic" value="$(arg objects_markers_topic)"/>
            <param name="objects_topic" value="$(arg objects_topic)"/>
            <param name="object_speed_threshold" value="$(arg object_speed_threshold)"/>
            <param name="arrow_speed_threshold" value="$(arg arrow_speed_threshold)"/>
            <param name="marker_display_duration" value="$(arg marker_display_duration)"/>
            <param name="label_color" value="$(arg label_color)"/>
            <param name="arrow_color" value="$(arg arrow_color)"/>
            <param name="hull_color" value="$(arg hull_color)"/>
            <param name="box_color" value="$(arg box_color)"/>
            <param name="centroid_color" value="$(arg centroid_color)"/>

        </node>
        



    </group>

</launch>